### PR TITLE
Rewrite Offering creation and editing

### PR DESCRIPTION
**Pull request by jrjohnson**
Improved user experience by:
- using the same learner group manager as we use for ILM learner groups
- Switch to duration for selecting end date instead of a date
- Validate more consistently with warning instead of flash messages

Improved developer experience by:
- Removing a lot of duplicate code between offering editor and offering manager
- Brining offering-form up to our current thinking on component design
- Adding more integration tests.

WIP:
- [x] Test creating new offering
- [x] Test editing offering
- [x] Test creating recurring offering
- [x] Test creating small group offering
- [x] Test creating recurring small group offering
- [x] Validate new look and experience.
- [x] Needs translations

@saschaben / @dartajax this is now ready to be battle tested creating all kinds of offerings and editing existing ones.  Please make sure I didn't miss any existing functionality and that everything works as you would expect.

@stopfstedt this code should be ready for review.  Functionally compete and tests are added.  Sorry for the giant PR.

### CODE REVIEW

**Comment by saschaben** [comment]
So far so good with testing functionally; <strike>have not yet reviewed as member of assigned student group</strike>.
some layout tweaks however would be good if we can do them:
newoffering

**Comment by saschaben** [comment]
just a thought, POV from others would be helpful:

offgroupsmgr

**Comment by jrjohnson** [comment]
@saschaben I made these adjustments with two notes:

- I Agree that the alignment of the selected learner groups was weird with that padding.  I put it back flush against the edge.
- All of our inputs are now 2.5em.  I favor consistency on this, can you take a look at the new user form and see if we should adjust those back down to 2em as well?

**Comment by saschaben** [comment]
V. Cool. 'Last Mile' comments and response:

- 2.5em inputs: I'm fine with consistency...but our search/filter inputs are actually 1.6em in height. either we need to reduce everything to 1.6em (aaggh no no no), or increase the search input fields to match.

- the recurring event "weeks" input needs the same padding between in and its label as did duration:
recurpadding

- what do you think about flipping recurring and location, for when "offering" is selected as opposed to "small group"? Right now, with location on the left, it makes the "recurring" block jump to the right when selected; if we flipped them, then Location would just magically appear....maybe:
reversepositions

- finally along the same lines as item #2 above, should we put Instructor above the group picker, so that it appears in the same real estate as location when flipped on? It already pushes everything down so I'm not sure that is an issue, especially if we fix the bug that keeps search results from closing after selection.
instructors

**Comment by jrjohnson** [comment]
Ready for review.  @saschaben - can you check the boxes you have tested up top?

**Comment by saschaben** [comment]
👍

File: app/components/offering-form.js [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+    let offerings = [];
+    offerings.push({startDate, endDate, room, learnerGroups, instructorGroups, instructors});
+    if (!makeRecurring) {
+      return offerings;
+    }
+
+    const userPickedDay = moment(startDate).day();
+    //convert strings to numbers use parseFloat because parseInt takes a second
+    //argument and gets thrown off by map sending that argument as the counter
+    const recurringDayInts = recurringDays.map(parseFloat).sort();
```
Comment (stopfstedt) [inline app/components/offering-form.js]: gross. but acceptable with the code comment above. thanks!

**Comment by saschaben** [comment]
fixes #1311

**Pull request merged**